edx_oauth2_provider/forms.py

```diff
@@ -107,9 +110,11 @@ def clean(self):
         try:
             client = Client.objects.get(client_id=data.get('client_id'))
         except Client.DoesNotExist:
+            log.exception("Invalid client ID '{}'.".format(data.get('client_id')))
```

Comment:
In order to more easily find the log statement, consider adding a prefix to the log message, such as: "OAuth2: ".